chainable_api.go

```diff
@@ -380,14 +380,68 @@ func (db *DB) Scopes(funcs ...func(*DB) *DB) (tx *DB) {
 }
 
 func (db *DB) executeScopes() (tx *DB) {
+	if len(db.Statement.scopes) == 0 {
+		return db
+	}
+
 	scopes := db.Statement.scopes
 	db.Statement.scopes = nil
+	originClause := db.Statement.Clauses
+
+	// use clean db in scope
+	cleanDB := db.Session(&Session{})
+	cleanDB.Statement.Clauses = map[string]clause.Clause{}
+
+	txs := make([]*DB, 0, len(scopes))
 	for _, scope := range scopes {
-		db = scope(db)
+		txs = append(txs, scope(cleanDB))
 	}
+
+	db.Statement.Clauses = originClause
+	db.mergeClauses(txs)
 	return db
 }
 
+func (db *DB) mergeClauses(txs []*DB) {
+	if len(txs) == 0 {
+		return
+	}
+
+	for _, tx := range txs {
+		stmt := tx.Statement
```

Comment:
Would it work if we make a function `mergeClause(tx *DB)` and then called it in the loop on line 397? `db.mergeClause(scope(cleanDB))`

Reply:
To keep the order of conditions, such as `.Where("a = ?",1).Scope(...)` the condition `a = 1` should come first

Reply:
This PR is to avoid the impact of multiple `AddClause` on the db instance, so a clean stmt needs to be cached (`Clauses` are not affected by `AddClause`)